pandas/core/indexes/category.py

```diff
@@ -872,6 +872,9 @@ def _delegate_method(self, name, *args, **kwargs):
             return res
         return CategoricalIndex(res, name=self.name)
 
+    def _is_compatible_with_other(self, other):
```

Comment:
does the super method not already do this? if not, can you show the case where it fails (which would be a bug)

Reply:
my point is the super method just not enough here? the dtypes must match exactly or we go to object.

Reply:
Ah I see what you're saying. I assumed we wanted to consider categorical indexes with different categories compatible. If not then yes the super method should be sufficient, I will make that change.